core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java

```diff
@@ -143,4 +143,19 @@ private void deleteRemovedActivityCategories(Set<ActivityCategory> activityCateg
 		activityCategoriesInRepository.stream().filter(ac -> !activityCategoryDTOsMap.containsKey(ac.getName()))
 				.forEach(this::deleteActivityCategory);
 	}
+
+	public Set<ActivityCategoryDTO> getMatchingActivityCategories(Set<String> smoothwallCategories)
```

Comment:
The method names ``getMatchingActivityCategories`` and ``getAppActivityCategories`` are inconsistent. I propose to name them ``getMatchingCategoriesForSmoothwallCategories`` and ``getMatchingCategoriesForApp``.